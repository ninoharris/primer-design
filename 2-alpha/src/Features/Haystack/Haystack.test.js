import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { Haystack } from './index'

const props = {
  "forward": "GTACACGGTGGTAACCCATAGGTCCGCTCCCCCAGCTTGGTGCATCAGGTCCTAATACAGCGGCCGGCTGCTGTAGGCCGTTGTAGAAACAGTAGACACGGGCACTCGGAACCTGCAGACCTCAGTGAGCTGAGTCGCCTTCTTTTGTCCCAAAGTGGCGTGCGCAGCACATGCCCCTCAGCGGGTGGCTATTACTGTCCAAGGGGGTCACACCCACCAATGTGCGACCGCGGAGCTGGACCACTTGTGCAAAGCTGTGCCACCCGACTTCGCTCGTGCAGGGTACGTGGGGCACAGCGA",
  "reverse": "CATGTGCCACCATTGGGTATCCAGGCGAGGGGGTCGAACCACGTAGTCCAGGATTATGTCGCCGGCCGACGACATCCGGCAACATCTTTGTCATCTGTGCCCGTGAGCCTTGGACGTCTGGAGTCACTCGACTCAGCGGAAGAAAACAGGGTTTCACCGCACGCGTCGTGTACGGGGAGTCGCCCACCGATAATGACAGGTTCCCCCAGTGTGGGTGGTTACACGCTGGCGCCTCGACCTGGTGAACACGTTTCGACACGGTGGGCTGAAGCGAGCACGTCCCATGCACCCCGTGTCGCT",
  "FG": "",
  "RG": "",
  "restrictionSites": [
    {
      "seq": "CTGCAG",
      "name": "PstI",
      "len": 6,
      "cutsForward": 5,
      "cutsReverse": 1,
      "color": "#aa4fc3",
      "pos": 112
    }, {
      "seq": "CCGCGG",
      "name": "SacII",
      "len": 6,
      "cutsForward": 4,
      "cutsReverse": 2,
      "color": "#67a77f",
      "pos": 227
    },
  ]
}

test('Haystack displays, example -L3d6TNs2PKt-fMf7_lS', () => {
  const renderer = new ShallowRenderer()
  renderer.render(<Haystack {...props} />)
  expect(renderer.getRenderOutput()).toMatchSnapshot()
})