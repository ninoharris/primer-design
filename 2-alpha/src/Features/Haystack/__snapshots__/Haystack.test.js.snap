// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haystack displays, example -L3d6TNs2PKt-fMf7_lS 1`] = `
<div
  className="haystack mt-3"
>
  <div
    style={
      Object {
        "position": "relative",
        "top": "-1rem",
      }
    }
  >
    <div
      className="forward"
    >
      <div
        className="multiline"
      >
        <div
          className="sequence"
        >
          <HelperPosition
            length={300}
          />
          <HaystackRestrictionSites
            restrictionSites={
              Object {
                "112": Object {
                  "color": "#aa4fc3",
                  "cutsForward": 5,
                  "cutsReverse": 1,
                  "name": "PstI",
                  "pos": 112,
                  "seq": "CTGCAG",
                },
                "227": Object {
                  "color": "#67a77f",
                  "cutsForward": 4,
                  "cutsReverse": 2,
                  "name": "SacII",
                  "pos": 227,
                  "seq": "CCGCGG",
                },
              }
            }
            seq="GTACACGGTGGTAACCCATAGGTCCGCTCCCCCAGCTTGGTGCATCAGGTCCTAATACAGCGGCCGGCTGCTGTAGGCCGTTGTAGAAACAGTAGACACGGGCACTCGGAACCTGCAGACCTCAGTGAGCTGAGTCGCCTTCTTTTGTCCCAAAGTGGCGTGCGCAGCACATGCCCCTCAGCGGGTGGCTATTACTGTCCAAGGGGGTCACACCCACCAATGTGCGACCGCGGAGCTGGACCACTTGTGCAAAGCTGTGCCACCCGACTTCGCTCGTGCAGGGTACGTGGGGCACAGCGA"
          />
          <Left5 />
          GTACACGGTGGTAACCCATAGGTCCGCTCCCCCAGCTTGGTGCATCAGGTCCTAATACAGCGGCCGGCTGCTGTAGGCCGTTGTAGAAACAGTAGACACGGGCACTCGGAACCTGCAGACCTCAGTGAGCTGAGTCGCCTTCTTTTGTCCCAAAGTGGCGTGCGCAGCACATGCCCCTCAGCGGGTGGCTATTACTGTCCAAGGGGGTCACACCCACCAATGTGCGACCGCGGAGCTGGACCACTTGTGCAAAGCTGTGCCACCCGACTTCGCTCGTGCAGGGTACGTGGGGCACAGCGA
          <Right3 />
          <Connect(HaystackForward) />
        </div>
      </div>
    </div>
    <div
      className="reverse"
    >
      <div
        className="multiline"
      >
        <div
          className="sequence"
        >
          <Connect(HaystackForward) />
          <Left3 />
          CATGTGCCACCATTGGGTATCCAGGCGAGGGGGTCGAACCACGTAGTCCAGGATTATGTCGCCGGCCGACGACATCCGGCAACATCTTTGTCATCTGTGCCCGTGAGCCTTGGACGTCTGGAGTCACTCGACTCAGCGGAAGAAAACAGGGTTTCACCGCACGCGTCGTGTACGGGGAGTCGCCCACCGATAATGACAGGTTCCCCCAGTGTGGGTGGTTACACGCTGGCGCCTCGACCTGGTGAACACGTTTCGACACGGTGGGCTGAAGCGAGCACGTCCCATGCACCCCGTGTCGCT
          <Right5 />
        </div>
        <Connect(Codons)
          seq="GTACACGGTGGTAACCCATAGGTCCGCTCCCCCAGCTTGGTGCATCAGGTCCTAATACAGCGGCCGGCTGCTGTAGGCCGTTGTAGAAACAGTAGACACGGGCACTCGGAACCTGCAGACCTCAGTGAGCTGAGTCGCCTTCTTTTGTCCCAAAGTGGCGTGCGCAGCACATGCCCCTCAGCGGGTGGCTATTACTGTCCAAGGGGGTCACACCCACCAATGTGCGACCGCGGAGCTGGACCACTTGTGCAAAGCTGTGCCACCCGACTTCGCTCGTGCAGGGTACGTGGGGCACAGCGA"
        />
      </div>
    </div>
  </div>
</div>
`;
